src/views/Validation.vue

```diff
@@ -0,0 +1,244 @@
+<template>
+  <div>
+    <h2>Validation</h2>
+
+    <div class="row">
+      <div class="col">
+        <p>
+          Reporting period {{ periodId }} has {{ uploads.length }} uploads.
+          {{ validUploads.length }} of these are currently valid.
+        </p>
+      </div>
+
+      <div class="col form-inline">
+        <div class="form-check">
+          <input class="form-check-input" type="checkbox" id="commitRevalidate" v-model="commit">
+          <label class="form-check-label" for="commitRevalidate">
+            Actually mark with updated validity?
+          </label>
+        </div>
+
+        <button class="btn ml-2" :class="revalidateBtnClass" @click="revalidate" :disabled="disableRevalidateBtn">
+          <div class="spinner-grow" role="status" v-if="revalidating">
+            <span class="sr-only">Loading...</span>
+          </div>
+
+          <span v-if="commit">Revalidate uploads</span>
+          <span v-else>Check upload validity</span>
+        </button>
```

Comment:
```suggestion
        <button class="btn ml-2" :class="revalidateBtnClass" @click="revalidate" :disabled="disableRevalidateBtn">
          <span v-if="revalidating" class="spinner-grow spinner-grow-sm" role="status">
          </span>
          <span v-if="revalidating"> Loading...</span>
          <span v-else>
            <span v-if="commit">Revalidate uploads</span>
            <span v-else>Check upload validity</span>
          </span>
        </button>
```

I think the layout is a little wonky here.  maybe something like this is cleaner?

Before:

![image](https://user-images.githubusercontent.com/11449340/169711910-155ace7e-8f6d-421e-8bb0-eabbab1c6b4f.png)

After:

![image](https://user-images.githubusercontent.com/11449340/169711895-bae7451b-b0a0-4c95-9ebe-b861037f6547.png)